timesketch/lib/analyzers/sequence_sessionizer_test.py

```diff
@@ -0,0 +1,430 @@
+        index = 'test_index'
+        sketch_id = 1
+        self.assertRaises(RuntimeError, sessionizer=\
+            NoneSessionTypeSequenceSessionizer(index, sketch_id))
+
+    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_event_seq_none(self):
+        """Test event_seq is not None."""
+        index = 'test_index'
+        sketch_id = 1
+        self.assertRaises(RuntimeError, sessionizer=\
+            NoneSeqSequenceSessionizer(index, sketch_id))
+
+    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_event_seq_empty(self):
+        """Test event_seq is not empty."""
+        index = 'test_index'
+        sketch_id = 1
+        self.assertRaises(RuntimeError, sessionizer=\
+            EmptySeqSequenceSessionizer(index, sketch_id))
+
+    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_no_timestamp(self):
+        """Test return_fields includes timestamp."""
+        index = 'test_index'
+        sketch_id = 1
+        self.assertRaises(RuntimeError, sessionizer=\
+            NoTimestampSequenceSessionizer(index, sketch_id))
+
+    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_no_return_attr(self):
+        """Test in return_fields are specified attributes."""
+        index = 'test_index'
+        sketch_id = 1
+        self.assertRaises(RuntimeError, sessionizer=\
+            NoReturnAttrSequenceSessionizer(index, sketch_id))
+
+    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_sessionizer(self):
+        """Test basic sequence sessionizer functionality."""
+        index = 'test_index'
+        sketch_id = 1
+        sessionizer = ManyEventsSequenceSessionizer(index, sketch_id)
+        self.assertIsInstance(sessionizer, SequenceSessionizerSketchPlugin)
+        self.assertEqual(index, sessionizer.index_name)
+        self.assertEqual(sketch_id, sessionizer.sketch.id)
+
+    @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
+                MockDataStore)
+    def test_one_session(self):
+        """Test one sequence of events is finded and allocated as a session."""
+        with mock.patch.object(SequenceSessionizerSketchPlugin,
+                               'event_stream',
+                               return_value=_create_mock_event(
+                                   0, 2, one_session_args, time_diffs=[1])):
+            index = 'test_index'
+            sketch_id = 1
+            sessionizer = ManyEventsSequenceSessionizer(index, sketch_id)
+            message = sessionizer.run()
+            self.assertEqual(
+                message,
+                'Sessionizing completed, number of {0:s} session created: 1'.
```

Comment:
Quick note: if you fix the string in the previous comments, you'll have to adjust this string too.

```suggestion
                'Sessionizing completed, number of {0:s} sessions created: 1'.
```